python-cache-demo/geo-service/geolocation_service.py

```diff
@@ -28,6 +23,9 @@
   database="geolocation"
 )
 
+# invalidate cache after 1 hour
+#TODO: should we add a check and invalidate cache once some data changes for the owners?
```

Comment:
Yeah so for this, i think we can just query all devices from database, and then if we get a device with an OwnerId that we currently don't have cached, then we invalidate the cache and get the owners-to-device data anew. The only weakness of this (in theory) is that we could have devices move between owners (cached) - that could be a blind spot but it's easier to just say that we don't support such "owner changes"